sdk/go/auto/local_workspace.go

```diff
@@ -460,6 +461,19 @@ func (l *LocalWorkspace) Stack(ctx context.Context) (*StackSummary, error) {
 	return nil, nil
 }
 
+// StackChangeSecretsProvider edits the secrets provider for the given stack.
+func (l *LocalWorkspace) StackChangeSecretsProvider(
+	ctx context.Context, stackName, newSecretsProvider, stdin string,
```

Comment:
> `stdin string`

I don't love calling this `stdin`. If eventually we move away from a CLI-backed implementation under the covers, I'm not sure what `stdin` even means anymore.

Also, what happens for the secret providers that don't need a value supplied to stdin? I assume I just pass an empty string in that case, but it's not immediately obvious from the parameter name that it's optional in that way.

I'm wondering if instead we should have something like:

```go
type ChangeStackSecretsProviderOptions struct {
	// NewPassphrase is the new passphrase when changing to a `passphrase` provider
	NewPassphrase *string
}

ChangeStackSecretsProvider(ctx context.Context, stackName, newSecretsProvider string,
	opts *ChangeStackSecretsProviderOptions) error
```

Where nil can be passed for the last parameter if not opts are needed.

Usage:

```go
s.Workspace().StackChangeSecretsProvider(ctx, s.Name(), "default", nil)

newPassphrase := "newpassphrase"
s.Workspace().StackChangeSecretsProvider(ctx, s.Name(), "passphrase", &auto.ChangeStackSecretsProviderOptions{
	NewPassphrase: &newPassphrase,
})
```

I could also imagine some alternative variants:

```go
// 1. Single method with opts
ChangeStackSecretsProvider(ctx context.Context, stackName, newSecretsProvider string,
	opts *ChangeStackSecretsProviderOptions) error

// 2. Single method with newPassphrase
ChangeStackSecretsProvider(ctx context.Context, stackName, newSecretsProvider string,
	newPassphrase *string) error

// 3. Multiple methods, one without opts, one with opts
ChangeStackSecretsProvider(ctx context.Context, stackName, newSecretsProvider string) error
ChangeStackSecretsProviderWithOptions(ctx context.Context, stackName, newSecretsProvider string,
	opts *ChangeStackSecretsProviderOptions) error

// 4. Single method with opts based on functional args pattern
ChangeStackSecretsProvider(ctx context.Context, stackName, newSecretsProvider string,
	opts ...optchangesecretsprovider.Option) error
```

1 or 2 would be my preference.